apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis_impl_producer.erl

```diff
@@ -94,11 +94,15 @@ on_stop(InstanceId, _State) ->
     ?status_connected
     | ?status_disconnected
     | {?status_disconnected, state(), {unhealthy_target, string()}}.
-on_get_status(_InstanceId, #{pool_name := Pool} = State) ->
+on_get_status(_InstanceId, #{pool_name := _Pool} = State) ->
+    do_get_status(State, []).
+
+-spec do_get_status(state(), nil() | [_Stream]) -> _.
+do_get_status(#{pool_name := Pool}, StreamArgs) ->
     case
         emqx_resource_pool:health_check_workers(
             Pool,
-            {emqx_bridge_kinesis_connector_client, connection_status, []},
+            {emqx_bridge_kinesis_connector_client, connection_status, StreamArgs},
```

Comment:
Q: so `[]` means check all channels?

Reply:
This was done only to please Elvis.  :sweat_smile: 

Essentially, the HC logic was the same, except that the resource HC (connector) doesn't have a specific stream to check, so it just tries to list streams: https://github.com/emqx/emqx/blob/ddbc493a3128cb10fd9f0aa8533febc5ae320093/apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis_connector_client.erl#L134-L142

Reply:
... but it doesn't actually test each one.  And when one stream is specified, it's checked differently:

https://github.com/emqx/emqx/blob/ddbc493a3128cb10fd9f0aa8533febc5ae320093/apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis_connector_client.erl#L131-L132

https://github.com/emqx/emqx/blob/ddbc493a3128cb10fd9f0aa8533febc5ae320093/apps/emqx_bridge_kinesis/src/emqx_bridge_kinesis_connector_client.erl#L166-L167